modules/cart/commerce_cart.module

```diff
@@ -103,6 +104,20 @@ function commerce_cart_commerce_order_delete(OrderInterface $order) {
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function commerce_cart_commerce_order_presave(OrderInterface $order) {
```

Comment:
Hm. So maybe it was unsetting the cart value, but the block was stale on just the checkout complete page?